### ISSUE TITLE
"Return type hint can be used" -> do not complain if parent method exists

### ISSUE THREAD

**Issue by donquixote** [kalessil/phpinspectionsea#706]
```php
interface I {

  /**
   * @return bool
   */
  public function isEnabled();
}

class C implements I {

  /**
   * @return bool
   */
  public function isEnabled() {
    return true;
  }
}
```
Currently, I get "': bool' can be declared as return type hint." both on I::isEnabled() and on C::isEnabled().
Maybe I want to retain the spirit of the parent method. Especially if the parent is in a separate 3rd party library.

## Proposed change

Introduce an option under "Return type hint can be used" to "Ignore, if parent method exists.".

Add a new inspection + fix "Add return type hint to comply with parent method", with bulk option.

The same can be done for parameters, see #702

## How does this help?

People can now do this:
1. Add return type hint on the interface method, without using refactor/rename. At first, this will break all implementations.
2. Run inspection.
3. Fix all the implementations at once.

**Comment by funivan** [comment]
> "Add return type hint to comply with parent method", with bulk option.
I do not think that this is an easy task. You need to analyze all child methods and detect that all return points is the same (returning same type)
This operation can decrease performance. Because you need to do this for all methods. But maybe I`m wrong. 

How I do that: go to parent method and call Refactoring action `Change signature`

**Comment by donquixote** [comment]
> I do not think that this is an easy task. You need to analyze all child methods and detect that all return points is the same (returning same type)

This inspection already exists. Just no bulk fix available.

**Comment by kalessil** [comment]
I need to check if this technically possible - quick-fixes can modify only the file it was attached to. 
Bulk operation will affect multiple files, let see if we can do this.

**Comment by kalessil** [comment]
Taking into account that #701 has been fixed and it will propose using `change signature` intention (when a method overrides/implements a parent one), I'll close this one.

`Change signature` should take care about complying with parent definition and taking care about children.

**Comment by donquixote** [comment]
@kalessil Just a thought I had.
Sometimes the parent method might be in a 3rd party package, or changing it would cause a BC break. So. in such cases a developer probably wants to silence this inspection.

**Comment by kalessil** [comment]
@donquixote : there are alternatives to silencing in this case (for future people: aggregation / Pull Request to the library are 2 great alternatives).

Since the inspection belongs to Language level migration group, I think we should keep current behaviour - I think this will pop up again and at that time we can come up with a better solution.

Just for protocol: I normally adding new settings without any questions, but in this case, I think proposed settings is not how we should address the issue.
